zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1313,7 +1354,7 @@ private void fireAllEvents(WatchedEvent event) {
    * are deleted before the last page is fetched. The upstream caller should be able to handle this.
    */
   public List<String> getChildren(String path) {
-    return getChildren(path, hasListeners(path));
+    return getChildren(path, (hasListeners(path) && !_usePersistWatcher));
```

Comment:
check `_usePersistWatcher` first, this should save some computation!

Reply:
Good suggestion! Updated